Sources/SwiftProtobuf/Message+AsyncMessageSequence.swift

```diff
@@ -0,0 +1,135 @@
+//
+// Sources/SwiftProtobuf/Message+AsyncMessageSequence.swift - Async sequence over binary delimited protobuf
+//
+// Copyright (c) 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Extensions to `Message` to provide an async sequence over a binary delimited protobuf stream.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+@available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
+extension Message {
+  
+  /// Creates an asynchronous sequence of messages decoded from resource bytes
+  public static func asyncSequence(
+    asyncBytes: URL.AsyncBytes,
+    extensions: ExtensionMap? = nil,
+    partial: Bool = false,
+    options: BinaryDecodingOptions = BinaryDecodingOptions()
+  ) -> AsyncMessageSequence<Self> {
+    let messageFactory = MessageFactory<Self>(
+      extensions: extensions,
+      partial: partial,
+      options: options
+    )
+    return AsyncMessageSequence(
+      asyncBytesIterator: asyncBytes.makeAsyncIterator(),
+      messageFactory: messageFactory
+    )
+  }
+}
+
+@available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
+public struct AsyncMessageSequence<M: Message> : AsyncSequence, Sendable {
+  
+  let asyncBytesIterator: URL.AsyncBytes.AsyncIterator
```

Comment:
This is a bit to specific for `AsyncBytes` what we want to have here is the `AsyncMessageSequence` to be generic over two things like this `AsyncMessageSequence<Base: AsyncSequence, M: Message>: AsyncSequence where Base.Element = UInt8`

Relatedly, we shouldn't store an iterator here but rather the base async sequence.

Reply:
Yes. Done.